hubclient/client.go

```diff
@@ -225,115 +225,107 @@ func (c *Client) httpGet(url string, result interface{}, expectedStatusCode int,
 
-	c.applyHeaderValues(req)
+	client.applyHeaderValues(req)
 	log.Debugf("PUT Request: %+v.", req)
 
 	httpStart := time.Now()
-	if resp, err = c.httpClient.Do(req); err != nil {
-		body := readResponseBody(resp, c.debugFlags)
+	var resp *http.Response
+	if resp, err = client.httpClient.Do(req); err != nil {
+		body := readResponseBody(resp, client.debugFlags)
 		return newHubClientError(body, resp, fmt.Sprintf("error getting HTTP Response from PUT to %s: %+v", url, err), err)
 	}
 
-	if c.debugFlags&HubClientDebugTimings != 0 {
+	if client.debugFlags&HubClientDebugTimings != 0 {
 		httpElapsed := time.Since(httpStart)
 		log.Debugf("DEBUG HTTP PUT ELAPSED TIME: %d ms.   -- Request: %s", (httpElapsed / 1000 / 1000), url)
 	}
 
-	return AnnotateHubClientErrorf(c.processResponse(resp, nil, expectedStatusCode), "unable to process response from PUT to %s", url) // TODO: Maybe need a response too?
+	return AnnotateHubClientErrorf(client.processResponse(resp, nil, expectedStatusCode), "unable to process response from PUT to %s", url) // TODO: Maybe need a response too?
 }
 
 func (c *Client) HttpPostString(url string, data string, contentType string, expectedStatusCode int) (string, error) {
-	return c.httpPost(url, data, contentType, expectedStatusCode, false)
-}
+	if c.debugFlags&HubClientDebugTimings != 0 {
+		log.Debugf("DEBUG HTTP STARTING %s STRING REQUEST: %s", http.MethodPost, url)
+	}
 
-func (c *Client) HttpPostJSON(url string, data interface{}, contentType string, expectedStatusCode int) (string, error) {
-	return c.httpPost(url, data, contentType, expectedStatusCode, true)
+	reader := strings.NewReader(data)
+	return postRequest(url, reader, contentType, c, expectedStatusCode)
 }
 
-func (c *Client) httpPost(url string, data interface{}, contentType string, expectedStatusCode int, isJsonData bool) (string, error) {
-
-	var resp *http.Response
-
+func (c *Client) HttpPostJSON(url string, data interface{}, contentType string, expectedStatusCode int) (string, error) {
 	if c.debugFlags&HubClientDebugTimings != 0 {
-		log.Debugf("DEBUG HTTP STARTING %s REQUEST: %s", http.MethodPost, url)
+		log.Debugf("DEBUG HTTP STARTING %s JSON REQUEST: %s", http.MethodPost, url)
 	}
 
-	var reader io.Reader
-
-	if isJsonData {
-		// Encode json
-		var buf bytes.Buffer
-		enc := json.NewEncoder(&buf)
-
-		if err := enc.Encode(&data); err != nil {
-			return "", newHubClientError(nil, nil, fmt.Sprintf("error encoding json: %+v", err), err)
-		}
-		reader = &buf
-	} else {
-		str, _ := data.(string)
-		reader = strings.NewReader(str)
+	// Encode json
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	if err := enc.Encode(&data); err != nil {
+		return "", newHubClientError(nil, nil, fmt.Sprintf("error encoding json: %+v", err), err)
 	}
+	reader := &buf
+
+	return postRequest(url, reader, contentType, c, expectedStatusCode)
+}
 
+func postRequest(url string, reader io.Reader, contentType string, client *Client, expectedStatusCode int) (string, error) {
```

Comment:
same as with `putRequest`
```suggestion
func (c *Client) postRequest(url string, reader io.Reader, contentType string, expectedStatusCode int) (string, error) {
```